shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/prepare/datasource/MySQLDataSourcePreparer.java

```diff
@@ -52,11 +57,19 @@ public void prepareTargetTables(final PrepareTargetTablesParameter parameter) {
     }
     
     private List<String> getCreateTableSQL(final PrepareTargetTablesParameter parameter) {
-        PipelineDDLGenerator generator = new PipelineDDLGenerator(PipelineContext.getContextManager());
+        PipelineDDLGenerator generator = new PipelineDDLGenerator();
+        DataSource dataSource = parameter.getDataSourceManager().getDataSource(parameter.getDataSourceConfig());
+        ShardingSphereMetaData metaData = PipelineContext.getContextManager().getMetaDataContexts().getMetaData();
```

Comment:
`DataSource dataSource = parameter.getDataSourceManager().getDataSource(parameter.getDataSourceConfig());` might return ShardingSphereDataSource, could we reference the original code `database.getResource().getDataSources().get(dataSourceName)` in PipelineDDLGenerator?

And also PostgreSQL and openGauss implementations.

You could refer to:
- PrepareTargetTablesParameter.tablesFirstDataNodes
- JobDataNodeLine, JobDataNodeEntry (logicTableName, dataNodes)
- DataNode (dataSourceName, tableName), e.g. ds_0.t_order_0